packages/http-server/src/server.ts

```diff
@@ -97,67 +110,80 @@ export const createServer = (operations: IHttpOperation[], opts: IPrismHttpServe
+            components.logger.warn({ name: 'VALIDATOR' }, message);
           } else {
-            request.log.info({ name: 'VALIDATOR' }, message);
+            components.logger.info({ name: 'VALIDATOR' }, message);
           }
         });
 
         return TE.fromIOEither(() =>
           E.tryCatch(() => {
-            if (output.headers) reply.headers(output.headers);
+            if (output.headers)
+              Object.entries(output.headers).forEach(([name, value]) => reply.setHeader(name, value));
 
-            reply
-              .code(output.statusCode)
-              .serializer((payload: unknown) => serialize(payload, reply.getHeader('content-type')))
-              .send(output.body);
+            send(
+              reply,
+              output.statusCode,
+              serialize(output.body, reply.getHeader('content-type') as string | undefined)
+            );
           }, E.toError)
         );
       }),
       TE.mapLeft((e: Error & { status?: number; additional?: { headers?: Dictionary<string> } }) => {
-        if (!reply.sent) {
-          const status = e.status || 500;
-          reply
-            .type('application/problem+json')
-            .serializer(JSON.stringify)
-            .code(status);
-
-          if (e.additional && e.additional.headers) {
-            reply.headers(e.additional.headers);
-          }
+        if (!reply.finished) {
+          reply.setHeader('content-type', 'application/problem+json');
 
-          reply.send(ProblemJsonError.fromPlainError(e));
+          if (e.additional && e.additional.headers)
+            Object.entries(e.additional.headers).forEach(([name, value]) => reply.setHeader(name, value));
+
+          send(
+            reply,
+            e.status || 500,
+            JSON.stringify(ProblemJsonError.fromPlainError(e))
+          );
         } else {
-          reply.res.end();
+          reply.end();
         }
 
-        request.log.error({ input }, `Request terminated with error: ${e}`);
+        components.logger.error({ input }, `Request terminated with error: ${e}`);
       })
     )();
   };
 
-  opts.cors
-    ? server.route({
-        url: '*',
-        method: ['GET', 'DELETE', 'HEAD', 'PATCH', 'POST', 'PUT'],
-        handler: replyHandler,
-      })
-    : server.all('*', replyHandler);
+  const server = micri(
+    Router.router(
+      Router.on.options(
+        () => opts.cors,
+        (req: IncomingMessage, res: ServerResponse) => {
+          res.setHeader('Access-Control-Allow-Origin', req.headers['origin'] || '*' as string);
+          res.setHeader('Access-Control-Allow-Credentials', 'true');
+          res.setHeader('Access-Control-Allow-Methods', 'GET,DELETE,HEAD,PATCH,POST,PUT');
+          send(res, 204);
+        },
+      ),
+      Router.otherwise(micriHandler)
+    )
+  );
+
+  const prism = createInstance(config, components);
 
-  const prismServer: IPrismHttpServer = {
+  return {
```

Comment:
I'm ok with this, however I recall I did this for a reason…but since I can't remember it, let's remove it.

Reply:
👍